encoding/thrift/envelope.go

```diff
@@ -55,3 +56,45 @@ func (ev disableEnvelopingProtocol) DecodeEnveloped(r io.ReaderAt) (wire.Envelop
 		Value: value,
 	}, err
 }
+
+type disableEnvelopingNoWireProtocol struct {
+	stream.Protocol
+
+	Type wire.EnvelopeType
+}
+
+func (evsp disableEnvelopingNoWireProtocol) Reader(r io.Reader) stream.Reader {
+	return disableEnvelopingReader{
+		Reader: evsp.Protocol.Reader(r),
+		Type:   evsp.Type,
+	}
+}
+
+func (evsp disableEnvelopingNoWireProtocol) Writer(w io.Writer) stream.Writer {
+	return disableEnvelopingWriter{
+		Writer: evsp.Protocol.Writer(w),
+	}
+}
+
+type disableEnvelopingReader struct {
+	stream.Reader
+
+	Type wire.EnvelopeType
+}
+
+func (evr disableEnvelopingReader) ReadEnvelopeBegin() (stream.EnvelopeHeader, error) {
+	return stream.EnvelopeHeader{
+		Name:  "",
+		Type:  evr.Type,
+		SeqID: 1,
+	}, nil
+}
+
+func (evr disableEnvelopingReader) ReadEnvelopeEnd() error { return nil }
+
+type disableEnvelopingWriter struct {
+	stream.Writer
+}
+
+func (evw disableEnvelopingWriter) WriteEnvelopeBegin(stream.EnvelopeHeader) error { return nil }
+func (evw disableEnvelopingWriter) WriteEnvelopeEnd() error                        { return nil }
```

Comment:
What are the reasons to have something different than `EncodeEnveloped` and `DecodeEnveloped` ? In this implementation we have a begin and end method (which is great ok), but what is the purpose of having them?
Note: I understand this probable comes from `stream.Reader`, but just trying to understand this choice.

Reply:
It's not much of a choice as much as it's inspired by Apache's `TProtocol`, https://pkg.go.dev/github.com/apache/thrift/lib/go/thrift#TProtocol